engine/storage/src/main/java/org/apache/cloudstack/storage/endpoint/DefaultEndPointSelector.java

```diff
@@ -276,7 +276,7 @@ public EndPoint select(DataObject object) {
         }
         if (object instanceof TemplateInfo) {
             TemplateInfo tmplInfo = (TemplateInfo)object;
-            if (store.getScope().getScopeType() == ScopeType.ZONE && store.getScope().getScopeId() == null && tmplInfo.getTemplateType() == TemplateType.SYSTEM) {
+            if (store.getScope().getScopeType() == ScopeType.ZONE && store.getScope().getScopeId() == null || tmplInfo.getTemplateType() == TemplateType.SYSTEM) {
```

Comment:
`store.getScope().getScopeType() == ScopeType.ZONE && store.getScope().getScopeId() == null` is always false, is it?